### PR TITLE
1 created reusable workflow for triggering jenkins job

### PR DESCRIPTION
**Pull request by ppsinghcalance**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -59,10 +72,13 @@ jobs:
...
     needs: build-docker
-    uses: Calance-US/calance-workflows/.github/workflows/deploy.yml@main
+    uses: Calance-US/calance-workflows/.github/workflows/deploy.yml@test
     with:
       repository_name: ${{ github.event.repository.name }}
       version: ${{ needs.build-docker.outputs.version }}
       cluster_environment: ${{ needs.build-docker.outputs.cluster_environment }}
-    secrets:  
+    secrets:
       namespace: ${{ secrets.NAMESPACE }}
```
Comment (agoyalcalance) [inline .github/workflows/build.yml]: let's put upper case parameter - NAMESPACE to maintain consistency.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -59,10 +72,13 @@ jobs:
 
   deploy-to-kubernetes:
     needs: build-docker
-    uses: Calance-US/calance-workflows/.github/workflows/deploy.yml@main
+    uses: Calance-US/calance-workflows/.github/workflows/deploy.yml@test
```
Comment (agoyalcalance) [inline .github/workflows/build.yml]: this will become main branch.

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -41,17 +41,25 @@ on:
...
+        required: true
+        description: 'Jenkins USER'
 
 jobs:
   trigger-jenkins-job:
     runs-on: ubuntu-latest
-    steps:                                
+    steps:
       - name: Trigger Jenkins Job
         uses: prempratap09/build-jenkins-job@master
```
Comment (agoyalcalance) [inline .github/workflows/deploy.yml:59]: this is still using repository from your personal namespace. it should utilise Calance-US repository.

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -41,17 +41,25 @@ on:
...
-#        type: string
+      JENKINS_URL:
+        required: true
+        description: 'Jenkins URL'
+      JENKINS_USER:
+        required: true
+        description: 'Jenkins USER'
+      JENKINS_TOKEN:
+        required: true
+        description: 'Jenkins USER'
```
Comment (agoyalcalance) [inline .github/workflows/deploy.yml:52]: the description should be `Jenkins TOKEN` not `Jenkins USER`.

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -41,17 +41,25 @@ on:
       namespace:
```
Comment (agoyalcalance) [inline .github/workflows/deploy.yml:41]: this namespace should also be in uppercase.

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -57,12 +70,15 @@ jobs:
...
-  deploy-to-kubernetes:
-    needs: build-docker
-    uses: Calance-US/calance-workflows/.github/workflows/deploy.yml@main
-    with:
-      repository_name: ${{ github.event.repository.name }}
-      version: ${{ needs.build-docker.outputs.version }}
-      cluster_environment: ${{ needs.build-docker.outputs.cluster_environment }}
-    secrets:  
-      namespace: ${{ secrets.NAMESPACE }}
+#  deploy-to-kubernetes:
```
Comment (agoyalcalance) [inline .github/workflows/build.yml:73]: remove commented out lines.

**Review by agoyalcalance — CHANGES_REQUESTED** [review]
@ppsinghcalance i understand this in draft state.. however still attaching my feedbacks.

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -1,11 +1,32 @@
...
+        description: 'Enter the application namespace:'
+      JENKINS_URL:
+        required: true
+        description: 'Jenkins URL'
+      JENKINS_USER:
+        required: true
+        description: 'Jenkins USER'
+      JENKINS_TOKEN:
+        required: true
+        description: 'Jenkins USER'
```
Comment (agoyalcalance) [inline .github/workflows/build.yml:16]: same the description should be Jenkins TOKEN.

File: .github/workflows/build.yml (hunk 1 above)
Comment (ppsinghcalance) [reply .github/workflows/build.yml]: changed the code

File: .github/workflows/build.yml (hunk 2 above)
Comment (ppsinghcalance) [reply .github/workflows/build.yml]: removed the code, using the same in public-repository-template

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (ppsinghcalance) [reply .github/workflows/deploy.yml]: changed to Calance-US repository.

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (ppsinghcalance) [reply .github/workflows/deploy.yml]: Changed to Jenkins Token

File: .github/workflows/deploy.yml (hunk 5 above)
Comment (ppsinghcalance) [reply .github/workflows/deploy.yml]: made it uppercase

File: .github/workflows/build.yml (hunk 6 above)
Comment (ppsinghcalance) [reply .github/workflows/build.yml]: done

File: .github/workflows/build.yml (hunk 7 above)
Comment (ppsinghcalance) [reply .github/workflows/build.yml]: removed the piece of code form build.yml

**Review by agoyalcalance — APPROVED** [review]

**Pull request merged**